pandas/tests/test_frame.py

```diff
@@ -8028,6 +8030,19 @@ def test_corr_int(self):
         df3.cov()
         df3.corr()
 
+    def test_corr_int_and_boolean(self):
+        tm._skip_if_no_scipy()
+
+        # when dtypes of pandas series are different
+        # then ndarray will have dtype=object,
+        # so it need to be properly handled
+        df = DataFrame({"a": [True, False], "b": [1, 0]})
+
+        expected = np.ones((2, 2), dtype=np.float64)
```

Comment:
construct an actual ``DataFrame`` here and use ``assert_frame_equal`` for comparison